zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/authorization/DbAuthorizationState.java

```diff
@@ -23,63 +24,82 @@ public class DbAuthorizationState implements AuthorizationState, MutableAuthoriz
 
   private final PersistedPermissions persistedPermissions = new PersistedPermissions();
 
-  private final DbString ownerKey;
-  private final DbString ownerType;
-  private final DbString resourceKey;
+  private final DbLong ownerKey;
   private final DbString resourceType;
-  private final DbCompositeKey<DbString, DbString> resourceCompositeKey;
-  private final DbCompositeKey<
-          DbCompositeKey<DbString, DbString>, DbCompositeKey<DbString, DbString>>
-      ownerAndResourceCompositeKey;
-  // owner key + owner type + resource key + resource type -> permission
+  private final DbString permissionType;
+  private final DbCompositeKey<DbString, DbString> resourceTypeAndPermissionTypeCompositeKey;
+  private final DbCompositeKey<DbLong, DbCompositeKey<DbString, DbString>>
+      ownerKeyAndResourceTypeAndPermissionTypeCompositeKey;
+  // owner key + resource type + permission type -> permissions
```

Comment:
❓ I think `permissions` here may be misleading, my understanding of this new structure is that its effectively:
```
{
  "ownerKey": "1234L",
  "resourceType": "process-definition",
  "permission": "CREATE",
  "resourceIdentifiers": {
    "bpmn-process-id:foo",
    "bpmn-process-id:bar"
  }
```

If my understanding is correct regarding the structure then I think `permissionType` should be `permission` and `permissions` should be something like `resourceIdentifiers`.

Reply:
I'll trust you regarding the naming as it's more your domain 👍